### DISCUSSION TITLE
Summeriser - not printing anything while test is Running.

### DISCUSSION THREAD

**Discussion by Jayrajgondaliya** [zalopay-oss/jmeter-grpc-request#77] [Q&A] [answered]
While Running test with jmeter-grpc-result, 

While test is running JMeter summarizer normally prints summary of what is going on at every 30 seconds something like below.

`summary +    227 in 00:00:18 =   12.7/s Avg:    73 Min:    20 Max:   589 Err:   113 (49.78%) Active: 1 Started: 1 Finished: 0
summary +    416 in 00:00:30 =   13.9/s Avg:    69 Min:    18 Max:   174 Err:   208 (50.00%) Active: 1 Started: 1 Finished: 0
summary =    643 in 00:00:48 =   13.4/s Avg:    71 Min:    18 Max:   589 Err:   321 (49.92%)`

so, we know what is going on.

But in case of jmeter-grpc. 
it does not print,  sometime it prints for 1 time and then stops.

Is there anything we need do to setup for summarizer, I tried changing JMeter Properties for summarizer, but it didn't work.

**Comment by huynhminhtan** [comment, marked as answer]
Hi @Jayrajgondaliya,

It could be due to your test script that you have using stop conditions like while loop (please check condition While Controller), read external file (as reading CSV check Rewind on end of list),...

For simplicity, you can run the example here [#running-the-examples](https://github.com/zalopay-oss/jmeter-grpc-request/tree/master/dist/example), please make sure you have configured the correct config as described [#configurations](https://github.com/zalopay-oss/jmeter-grpc-request#configurations). Finally, you can get the latest version of JMeter gRPC Request, by checking the master branch then build and try again.

I hope everything will be all right.
Regards

**Reply by Jayrajgondaliya** [reply]
I was using TimeShappingTime for Dynamically adjusting thread counts to achieve Desired Throughput rate in testing, 
In that We have Target Concurrency: ${__tstFeedback(tstTimer,10,40,10)}  >> (starting thread, Max thread, Number of spare threads).

So, the number of spare threads I was using 80% of max thread, but if I reduce it down to 1 thread only, it is printing all summary.
